cmake/Modules/FindMKLDNN.cmake

```diff
@@ -41,28 +37,15 @@ ENDIF(NOT IDEEP_INCLUDE_DIR OR NOT MKLDNN_INCLUDE_DIR)
 LIST(APPEND MKLDNN_INCLUDE_DIR ${IDEEP_INCLUDE_DIR})
 
 IF(MKL_FOUND)
+  # Configure MKL-DNN
+  ADD_DEFINITIONS(-DUSE_MKL)
```

Comment:
This line seems deeply questionable, for a number of reasons:

1. Is this for MKL-DNN? `USE_MKL` isn't even used by any source files in MKL-DNN. As far as I can tell, it is only used in `os_blas.h`, and that header is only included by headers (and no source files).
2. Is this for Caffe2? There aren't any uses of `USE_MKL` in Caffe2 either; we refer to it as `CAFFE2_USE_MKL`
3. Is it for fbgemm??? That seems super weird.

Reply:
@ezyang 
Yes. USE_MKL is for MKL-DNN build. The flag is used to enable 

> https://github.com/pytorch/pytorch/pull/15387/files#diff-d25356f3acd786db9e5a28c08ced1235R5
> 
> This comment is out of date and should be deleted.

Yes. I'll remove this comments line soon.

Reply:
> https://github.com/pytorch/pytorch/pull/15387/files#diff-d25356f3acd786db9e5a28c08ced1235R5
> 
> This comment is out of date and should be deleted.

Yes. I'll remove this comments line soon.

Reply:
@ezyang 
The USE_MKL is set for MKL-DNN build only.
It's related to the detail configuration of MKL-DNN.
Please refer to below comments copied from MKL-DNN src.
Currently, with better perf for caffe2, we set the config as, USE_MKL=ON && USE_CBLAS=OFF

 *  USE_MKL  USE_CBLAS effect
 *  -------  --------- ------
 *  yes      yes       normal compile: jit *may* be preferred over Intel(R) MKL CBLAS
 *  yes      no        jit calls OK; assert if cblas is ever called
 *  no       yes       system-dependent CBLAS
 *  no       no        gemm convolution (or other blas) N/A; create stubs